src/RestControllers/FHIR/FhirPatientRestController.php

```diff
@@ -9,119 +9,109 @@
  * @copyright Copyright (c) 2018 Jerry Padgett <sjpadgett@gmail.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
-
+ 
 namespace OpenEMR\RestControllers\FHIR;
 
 use OpenEMR\Services\FHIR\FhirResourcesService;
 use OpenEMR\Services\FHIR\FhirPatientService;
 use OpenEMR\Services\FHIR\FhirValidationService;
 use OpenEMR\RestControllers\RestControllerHelper;
 use OpenEMR\FHIR\R4\FHIRResource\FHIRBundle\FHIRBundleEntry;
+use OpenEMR\Validators\ProcessingResult;
+
+require_once('_rest_config.php');
 
+/**
+ * Supports REST interactions with the FHIR patient resource
+ */
 class FhirPatientRestController
 {
     private $fhirPatientService;
     private $fhirService;
     private $fhirValidate;
 
-    public function __construct($pid)
+    public function __construct()
     {
         $this->fhirService = new FhirResourcesService();
         $this->fhirPatientService = new FhirPatientService();
-        $this->fhirPatientService->setId($pid);
         $this->fhirValidate = new FhirValidationService();
     }
 
+    /**
+     * Creates a new FHIR patient resource
+     * @param $fhirJson The FHIR patient resource
+     * @returns 201 if the resource is created, 400 if the resource is invalid
+     */
     public function post($fhirJson)
     {
         $fhirValidate = $this->fhirValidate->validate($fhirJson);
         if (!empty($fhirValidate)) {
-            return RestControllerHelper::responseHandler($fhirValidate, null, 404);
+            return RestControllerHelper::responseHandler($fhirValidate, null, 400);
         }
-        $data = $this->fhirPatientService->parsePatientResource($fhirJson);
```

Comment:
all of the parsing, validating (most of  the validating) occurs in the service layer now, which makes the controller a bit simpler.